src/main/java/nextstep/member/application/GithubClient.java

```diff
@@ -0,0 +1,80 @@
+package nextstep.member.application;
+
+import nextstep.member.application.dto.GithubAccessTokenRequest;
+import nextstep.member.application.dto.GithubAccessTokenResponse;
+import nextstep.member.application.dto.GithubProfileResponse;
+import nextstep.member.exception.ApiCallException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import static nextstep.common.constant.ErrorCode.GITHUB_NOT_FOUND;
+
+@Component
+public class GithubClient {
+    @Value("${security.github.access-token-uri}")
+    private String githubAccessTokenUrl;
+    @Value("${security.github.access-profile-uri}")
+    private String githubAccessProfileUrl;
+    @Value("${security.github.client-id}")
+    private String githubClientId;
+    @Value("${security.github.client-secret}")
+    private String githubClientSecret;
+
+    private final RestTemplate restTemplate;
+
+    public GithubClient(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public String requestGithubAccessToken(String code) {
+        GithubAccessTokenRequest githubAccessTokenRequest = new GithubAccessTokenRequest(
+                code,
+                githubClientId,
+                githubClientSecret
+        );
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Accept", MediaType.APPLICATION_JSON_VALUE);
+
+        HttpEntity<MultiValueMap<String, String>> httpEntity = new HttpEntity(
+                githubAccessTokenRequest, headers);
+
+
+        try {
+            ResponseEntity<GithubAccessTokenResponse> responseEntity = restTemplate
+                    .exchange(githubAccessTokenUrl, HttpMethod.POST, httpEntity, GithubAccessTokenResponse.class);
```

Comment:
Nit: `postForEntity()`를 활용해볼 수도 있습니다.
```suggestion
                    .postForEntity(githubAccessTokenUrl, httpEntity, GithubAccessTokenResponse.class);
```